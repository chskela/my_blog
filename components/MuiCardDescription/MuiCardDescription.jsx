import React from "react";
import PropTypes from "prop-types";

import classes from "./class.module.css";

import { formatDate } from "../../lib/formatDate";

const MuiCardDescription = React.memo(
  ({ category, title, author, date, excerpt }) => {
    const postDate = formatDate(date);
    return (
      <div className={classes.content}>
        <p data-role="content-category">{category}</p>
        <h1 data-role="content-title">{title}</h1>
        <div data-role="content-avatar">
          <p>{author.name}</p>
          <time>{postDate}</time>
        </div>
        <p data-role="content-excerpt">{excerpt}</p>
      </div>
    );
  }
);

MuiCardDescription.propTypes = {
  category: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.shape({
    name: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
  }),
  date: PropTypes.string,
  excerpt: PropTypes.string.isRequired,
};

MuiCardDescription.defaultProps = {
  date: "",
};

MuiCardDescription.displayName = "MuiCardDescription";

export default MuiCardDescription;
