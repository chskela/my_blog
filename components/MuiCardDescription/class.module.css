.content {
  display: flex;
  flex-direction: column;
}

.content > p[data-role="content-category"] {
  font-size: var(--size-5);
  color: var(--color-secondary-500);
  text-transform: uppercase;
  line-height: var(--size-7);
}

.content > h1[data-role="content-title"] {
  font-size: var(--size-8);
  margin: 0;
  margin-bottom: var(--size-3);
  line-height: var(--size-10);
}

.content > div[data-role="content-avatar"] {
  color: var(--color-secondary-500);
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: baseline;
  margin-bottom: var(--size-3);
}

.content > div[data-role="content-avatar"] > p {
  margin-right: var(--size-6);
}

.content > p[data-role="content-excerpt"] {
  overflow: hidden;
  line-height: 24px;
}
