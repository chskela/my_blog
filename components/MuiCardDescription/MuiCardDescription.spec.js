import React from "react";
import { shallow } from "enzyme";
import MuiCardDescription from "./MuiCardDescription";

const setUp = (props) => shallow(<MuiCardDescription {...props} />);

describe("MuiCardDescription component", () => {
  const props = {
    category: "tets",
    title: "test",
    author: { name: "test", picture: "test" },
    excerpt: "test",
  };

  it("should render MuiCardDescription component default props", () => {
    const component = setUp(props);
    expect(component).toMatchSnapshot();
  });
  it("should render MuiCardDescription component whis props", () => {
    const component = setUp({ ...props, date: "test" });
    expect(component).toMatchSnapshot();
  });
});
