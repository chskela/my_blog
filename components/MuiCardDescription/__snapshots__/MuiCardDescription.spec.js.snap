// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiCardDescription component should render MuiCardDescription component default props 1`] = `
<div
  className="content"
>
  <p
    data-role="content-category"
  >
    tets
  </p>
  <h1
    data-role="content-title"
  >
    test
  </h1>
  <div
    data-role="content-avatar"
  >
    <p>
      test
    </p>
    <time />
  </div>
  <p
    data-role="content-excerpt"
  >
    test
  </p>
</div>
`;

exports[`MuiCardDescription component should render MuiCardDescription component whis props 1`] = `
<div
  className="content"
>
  <p
    data-role="content-category"
  >
    tets
  </p>
  <h1
    data-role="content-title"
  >
    test
  </h1>
  <div
    data-role="content-avatar"
  >
    <p>
      test
    </p>
    <time>
      test
    </time>
  </div>
  <p
    data-role="content-excerpt"
  >
    test
  </p>
</div>
`;
